packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2177,29 +2177,27 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
   final List<_StandardBottomSheet> _dismissedBottomSheets = <_StandardBottomSheet>[];
   PersistentBottomSheetController<dynamic>? _currentBottomSheet;
   final GlobalKey _currentBottomSheetKey = GlobalKey();
+  LocalHistoryEntry? _persistentSheetHistoryEntry;
 
   void _maybeBuildPersistentBottomSheet() {
     if (widget.bottomSheet != null && _currentBottomSheet == null) {
       // The new _currentBottomSheet is not a local history entry so a "back" button
       // will not be added to the Scaffold's appbar and the bottom sheet will not
       // support drag or swipe to dismiss.
       final AnimationController animationController = BottomSheet.createAnimationController(this)..value = 1.0;
-      LocalHistoryEntry? persistentSheetHistoryEntry;
       bool persistentBottomSheetExtentChanged(DraggableScrollableNotification notification) {
-        if (notification.extent > notification.initialExtent) {
-          if (persistentSheetHistoryEntry == null) {
-            persistentSheetHistoryEntry = LocalHistoryEntry(onRemove: () {
-              if (notification.extent > notification.initialExtent) {
```

Comment:
This condition is always true, right?

Reply:
It would seem so.. I am out today and will take a closer look a little later. FYI @thkim1011, who is joining us in scrolling land. Tae has been studying the draggable scrollable sheet. :)

Reply:
@thkim1011 Hi, nice to meet you : )

Reply:
Isn't it true that, by definition persistent bottom sheets on `Scaffold.bottomSheet` should not create a `LocalHistoryEntry` as the sheet is considered part of the main UI as per material guidelines, from `Scaffold.bottomSheet` documentation: `Unlike the persistent bottom sheet displayed by showBottomSheet this bottom sheet is not a LocalHistoryEntry and cannot be dismissed with the scaffold appbar's back button`. If so why is it even creating a `LocalHistoryEntry`

Reply:
This statement in the scope of  `onRemove` callback and it should be never changed until it actually called, right?

Reply:
Line #2189 already ensures that this condition is true.

Reply:
Oh silly me, I missed that.